### PR TITLE
refactor: merge DataCollection and protobuf Node in spin/analytics

### PR DESCRIPTION
**Pull request by laipogo**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior?** (You can also refer to a JIRA ticket here)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)

* **Description of changes**

---

* **Please check if the PR fulfills these requirements**
- [ ] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [ ] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [ ] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [ ] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [ ] Code changes have run through `go fmt`
- [ ] Code changes have run through `go mod tidy`
- [ ] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

File: spin/analytics.go [hunk 1]
```diff
@@ -239,54 +218,28 @@ func (dc *dataCollection) doPrepData(btfsNode *core.IpfsNode) (*pb.SignedMetrics
...
-	nd.OsType = dc.OSType
-	nd.PeersConnected = dc.NumPeers
-	nd.StorageUsed = dc.StorageUsed
-	nd.UpTime = dc.UpTime
-	nd.Upload = dc.Upload
-	nd.TotalUpload = dc.TotalUp
-	nd.TotalDownload = dc.TotalDown
-	nd.HVal = dc.HVal
+func (dc *dcWrap) getPayload(btfsNode *core.IpfsNode) ([]byte, error) {
 	if config, err := dc.node.Repo.Config(); err == nil {
```
Comment (taiyangc) [inline spin/analytics.go]: config does not change without restarting daemon, so you can safely set this inside `Analytics()`

Comment (laipogo) [reply spin/analytics.go]: okay

File: spin/analytics.go [hunk 2]
```diff
@@ -74,6 +75,12 @@ func Analytics(node *core.IpfsNode, BTFSVersion, hValue string) {
 	dc := new(dcWrap)
 	dc.node = node
 	dc.pn = new(nodepb.Node)
+	if c, err := dc.node.Repo.Config(); err != nil {
```
Comment (taiyangc) [inline spin/analytics.go]: `configuration` was already read above, no need to get again.

File: spin/analytics.go [hunk 3]
```diff
@@ -227,10 +234,8 @@ func (dc *dcWrap) doSendData(ctx context.Context, config *config.Config, sm *pb.
 }
 
 func (dc *dcWrap) getPayload(btfsNode *core.IpfsNode) ([]byte, error) {
-	if config, err := dc.node.Repo.Config(); err == nil {
-		if storageMax, err := humanize.ParseBytes(config.Datastore.StorageMax); err == nil {
-			dc.pn.StorageVolumeCap = storageMax
-		}
+	if storageMax, err := humanize.ParseBytes(dc.config.Datastore.StorageMax); err == nil {
+		dc.pn.StorageVolumeCap = storageMax
```
Comment (taiyangc) [inline spin/analytics.go]: the whole line can be in analytics

**Comment by laipogo** [comment]
thanks, updated per comments.

**Pull request merged**